bundles/org.openhab.binding.neohub/src/main/java/org/openhab/binding/neohub/internal/NeoHubHandler.java

```diff
@@ -0,0 +1,247 @@
+    private NeoHubSocket socket;
+
+    private ScheduledFuture<?> lazyPollingScheduler;
+    private ScheduledFuture<?> fastPollingScheduler;
+
+    private final AtomicInteger fastPollingCallsToGo = new AtomicInteger();
+
+    public NeoHubHandler(Bridge bridge) {
+        super(bridge);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        // future: currently there is nothing to do for a NeoHub
+    }
+
+    @Override
+    public void initialize() {
+        config = getConfigAs(NeoHubConfiguration.class);
+
+        if (config == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_PENDING,
+                    "missing configuration, status => offline!");
+            return;
+        }
+
+        if (logger.isDebugEnabled())
+            logger.debug("hostname={}", config.hostName);
+
+        if (config.hostName.isEmpty()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "missing host name, status => offline!");
+            return;
+        }
+
+        if (logger.isDebugEnabled())
+            logger.debug("port={}", config.portNumber);
+
+        if (config.portNumber <= 0 || config.portNumber > 0xFFFF) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "port out of range, status => offline!");
+            return;
+        }
+
+        if (logger.isDebugEnabled())
+            logger.debug("polling interval={}", config.pollingInterval);
+
+        if (config.pollingInterval < FAST_POLL_INTERVAL || config.pollingInterval > LAZY_POLL_INTERVAL) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    String.format("polling interval out of range [%d..%d], status => offline!", FAST_POLL_INTERVAL,
+                            LAZY_POLL_INTERVAL));
+            return;
+        }
+
+        socket = new NeoHubSocket(config.hostName, config.portNumber);
+
+        if (logger.isDebugEnabled())
+            logger.debug("start background polling..");
+
+        // create a "lazy" polling scheduler
+        if (lazyPollingScheduler == null || lazyPollingScheduler.isCancelled()) {
+            lazyPollingScheduler = scheduler.scheduleWithFixedDelay(this::lazyPollingSchedulerExecute,
+                    config.pollingInterval, config.pollingInterval, TimeUnit.SECONDS);
+        }
+
+        // create a "fast" polling scheduler
+        fastPollingCallsToGo.set(FAST_POLL_CYCLES);
+        if (fastPollingScheduler == null || fastPollingScheduler.isCancelled()) {
+            fastPollingScheduler = scheduler.scheduleWithFixedDelay(this::fastPollingSchedulerExecute,
+                    FAST_POLL_INTERVAL, FAST_POLL_INTERVAL, TimeUnit.SECONDS);
+        }
+
+        // start a fast polling burst to ensure the NeHub is initialized quickly
+        startFastPollingBurst();
+    }
```

Comment:
You must set the ThingStatus at the end of this method. As you don't know whether you can reach the hub yet, best value is probably UNKNOWN.

Reply:
Ok, done.